ttnn/cpp/ttnn/operations/data_movement/transpose/device/transpose_program_factory.cpp

```diff
@@ -1776,8 +1776,9 @@ operation::ProgramWithCallbacks transpose_wh_multi_core_sharded(const Tensor& a,
     uint32_t dst_single_tile_size = tt::tt_metal::detail::TileSize(dst_cb_data_format);
 
     tt::tt_metal::Buffer* src0_buffer = a.buffer();
-
-    int32_t num_tiles = a.volume() / TILE_HW;
+    const auto tile = a.get_tensor_spec().tile();
+    const uint32_t tile_hw = tile.get_tile_hw();
+    int32_t num_tiles = a.volume() / tile_hw;
```

Comment:
Would be nice to simply ask num_tiles from the tensor spec? There is no such api right now. So the question is if you want it to be there

CC @sminakov-tt

Reply:
yes that would be great, this pattern is repeated everywhere